internal/config/ociartifact/ociartifact.go

```diff
@@ -121,15 +135,79 @@ func (o *OCIArtifact) Pull(ctx context.Context, img string, opts *PullOptions) (
 		return nil, fmt.Errorf("read with limit: %w", err)
 	}
 
-	if err := verifyDigest(&layer, layerBytes); err != nil {
+	if err := verifyDigest(layer, layerBytes); err != nil {
 		return nil, fmt.Errorf("verify digest of layer: %w", err)
 	}
 
+	keyPath := o.tryWriteCache(ctx, opts, layer, layerBytes)
+
 	return &Artifact{
 		Data: layerBytes,
+		Path: keyPath,
 	}, nil
 }
 
+func (o *OCIArtifact) tryLookupCache(ctx context.Context, opts *PullOptions, layer *manifest.LayerInfo) *Artifact {
+	if opts.CachePath == "" {
+		return nil
+	}
+
+	if err := o.impl.MkdirAll(opts.CachePath, 0o700); err != nil {
+		log.Errorf(ctx, "Unable to create cache path: %v", err)
+		// Unset cache path to avoid write.
+		opts.CachePath = ""
+		return nil
+	}
+
+	key, keyPath := keyAndPath(opts, layer)
+	tryRemoveKeyPath := false
+
+	if value, err := o.impl.ReadFile(key); err == nil {
+		if err := verifyDigest(layer, value); err == nil {
+			log.Infof(ctx, "Using cached artifact layer for digest %q", layer.BlobInfo.Digest)
+			return &Artifact{
+				Data: value,
+				Path: keyPath,
+			}
+		}
+
+		log.Warnf(ctx, "Unable to verify cached digest (removing it): %v", err)
+		tryRemoveKeyPath = true
+	} else if !os.IsNotExist(err) {
+		log.Warnf(ctx, "Unable to read cache key path (removing it): %v", err)
+		tryRemoveKeyPath = true
+	}
+
+	if tryRemoveKeyPath {
+		if err := o.impl.RemoveAll(keyPath); err != nil {
```

Comment:
```suggestion
		log.Infof(ctx, "Removing cached artifact layer for digest %q", layer.BlobInfo.Digest)
		if err := o.impl.RemoveAll(keyPath); err != nil {
```

If the above messages are updated, some paper trail will be in order.